R/render_site.R

```diff
@@ -220,19 +220,36 @@ clean_site <- function(input = ".", preview = FALSE, quiet = FALSE,
   # get the files to be cleaned
   files <- generator$clean()
 
+
+  if (length(files) == 0) {
+    if (preview || !quiet) cat("Nothing to removed. All clean !\n")
+    return(invisible(NULL))
+  }
+
   # if it's just a preview then return the files, otherwise
   # actually remove the files
-  if (preview)
-    files
-  else {
+  if (preview) {
+    cat("These files and folders can probably be removed:\n",
+        paste0("* ", mark_dirs(files)),
+        "\nUse `rmarkdown::clean_site(preview = FALSE)` to remove them.",
+        sep = "\n")
+  } else {
     if (!quiet) {
-      cat("Removing files: \n")
-      cat(paste0(" ", files), sep = "\n")
+      cat("Removing files: \n",
+          paste0("* ", mark_dirs(files)),
+          sep = "\n")
     }
     unlink(file.path(input, files), recursive = TRUE)
   }
 }
 
+# TODO: Move to xfun - bookdown use it too.
+mark_dirs <- function(files) {
+  i = file_test("-d", files) & !grepl("/$", files)
+  files[i] = paste0(files[i], "/")
```

Comment:
```suggestion
  i <- file_test("-d", files) & !grepl("/$", files)
  files[i] <- paste0(files[i], "/")
```

Reply:
arg... copy paste from xfun PR. Change in style guide is always tricky